File: lib1/src/main/java/com/huazhen/lib1/MainActivity.java
package com.huazhen.lib1;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.huazhen.services.ToastServices;

import cn.fxlcy.component.services.AppServices;

/**
 * Created by fxlcy on 18-7-17.
 */
@Route(path = "/path/main")
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ToastServices services = AppServices.getInstance().getService(ToastServices.class);

        services.toast(this, "sdfosdjfojsodfjosjdf");
    }
}

File: app/src/main/java/com/huazhen/component/MainActivity.kt
package com.huazhen.component

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.alibaba.android.arouter.facade.Postcard
import com.alibaba.android.arouter.facade.callback.NavigationCallback
import com.alibaba.android.arouter.launcher.ARouter

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)



        ARouter.init(application)
//        ARouter.openDebug()


        ARouter.getInstance().build("/path/main")
                .navigation(this,object :NavigationCallback{
                    override fun onLost(postcard: Postcard?) {

                    }

                    override fun onFound(postcard: Postcard?) {

                    }

                    override fun onInterrupt(postcard: Postcard?) {

                    }

                    override fun onArrival(postcard: Postcard?) {
                    }

                })

    }
}
